designs/modify-resources-attached-to-code-pipeline.md

```diff
@@ -0,0 +1,209 @@
+
+Metrics are logged indicating whether or not the Toolkit was successful in determining if there was an associated pipeline - see [Fallback Handling](#fallback-handling).
+
+### Utility Methods
+* a method that determines if an AWS resource is associated with AWS CodePipeline
+* a method that generates confirmation prompt text (if confirmation prompt will exist within a Wizard and as a standalone prompt)
+* a method that encapsulates showing users the (non-wizard) confirmation prompt
+
+### Confirmation Prompt
+
+The confirmation prompt can be implemented using one of two UI facilities:
+* `showInformationMessage`, which pops up a dismissable toast message that can contain buttons
+  * Pros
+    * Compact
+    * "Looks like" a conventional dialog box
+  * Cons
+    * if users press `Esc` on the toast, the toast hides in a queue instead of getting dismissed.
+      * it appears to the user that the flow has stopped ("I don't see the toast anymore") but the flow does not return to the calling code (calling code is actually waiting for a button to be pressed)
+    * Putting a third button along the lines of `More Info...` beside `Yes` and `No` buttons is unconventional
+    * a `More Info...` button dismisses the toast (and would have to be handled as a `No`), ideally the toast would stay open
+* either a `QuickPick` (which presents a list to the user for selection) or `InputBox` (which prompts the user for text). Both are capable of being complemented with buttons that allow for additional actions.
+  * Pros
+    * if users press `Esc` on the QuickPick/InputBox, the QuickPick/InputBox is dismissed, and flow returns to the calling code, which in this case can handle it as an intent to cancel
+    * More consistent UX if the prompt is used both as a standalone prompt, and in a wizard (CloudFormation Stack Deploy alternate proposal uses a wizard)
+    * Flexability to support actions (like bringing up a help page) while leaving the prompt open
+    * idiomatic to VS Code - using a QuickPick, the `Yes` and `No` choices appear in a selection list
+  * Cons
+    * None
+
+`QuickPick/InputBox` is the appropriate facility to use for the confirmation prompt. The first con listed in the toast approach would cause misleading experiences while accumulating promises that wait for additional action.
+
+### API Client
+
+The `ResourceGroupsTaggingAPI` service client will be used. The client will be set up in a manner consistent with the Lambda and CloudFormation clients, allowing the clients to be stubbed out in unit tests. See the Toolkit's Lambda Client [Interface](/src/shared/clients/lambdaClient.ts) and [Implementation](/src/shared/clients/defaultLambdaClient.ts) as an example. A new client factory will be added to [ToolkitClientBuilder](/src/shared/clients/toolkitClientBuilder.ts) and [DefaultToolkitClientBuilder](/src/shared/clients/defaultToolkitClientBuilder.ts).
```

Comment:
Are we only using the ResourceGroupsTaggingAPI or are we using a individual resource getters to check tags related to resource specific calls as well?